server/errorhandler.go

```diff
@@ -0,0 +1,47 @@
+package server
+
+import (
+	"io"
+	"net"
+	"net/http"
+
+	"github.com/containous/traefik/middlewares"
+)
+
+// RecordingErrorHandler is an error handler, implementing the vulcand/oxy
+// error handler interface, which is recording network errors by setting the
+// request context value with the key middlewares.DefaultNetErrCtxKey to true.
+// This enables the retry middleware to only retry on actual network errors.
+type RecordingErrorHandler struct{}
+
+// NewRecordingErrorHandler creates and returns a new instance of RecordingErrorHandler.
+func NewRecordingErrorHandler() *RecordingErrorHandler {
```

Comment:
As the implementation of `NewRecordingErrorHandler` isn't doing anything besides returning a pointer to an empty struct, couldn't we just drop it and have callers do it instead?